ompi/mca/coll/base/coll_base_bcast.c

```diff
@@ -684,9 +643,10 @@ ompi_coll_base_bcast_intra_basic_linear(void *buff, int count,
      * the error after we free everything. */
 
     err = ompi_request_wait_all(i, reqs, MPI_STATUSES_IGNORE);
-
-    /* Free the reqs */
-    ompi_coll_base_free_reqs(reqs, i);
+ err_hndl:
+    if( MPI_SUCCESS != err ) {  /* Free the reqs */
+        ompi_coll_base_free_reqs(reqs, i);
```

Comment:
@bosilca 
You can come to err_hndl if get_reqs() has failed, i.e. reqs=NULL.
So unless you insert this test inside free_reqs(), you should check the pointer before calling free_reqs()